demo/src/main/java/com/example/demo/services/impl/ClientServiceImpl.java

```diff
@@ -1,52 +1,101 @@
+        readClientDTO.setLastName(client.getLastName());
+        readClientDTO.setAge(client.getAge());
+        readClientDTO.setLogin(client.getLogin());
+        readClientDTO.setPassword(client.getPassword());
+        readClientDTO.setCreditBalance(client.getCreditBalance());
+        return readClientDTO;
     }
 
     @Override
-    public Client update(Client client, Long id) {
-        Client excitedClient = clientRepository.findById(id).orElse(null);
-        if(excitedClient != null){
-            excitedClient.setFirstName(client.getFirstName());
-            excitedClient.setLastName(client.getLastName());
-            excitedClient.setAge(client.getAge());
-            excitedClient.setLogin(client.getLogin());
-            excitedClient.setPassword(client.getPassword());
-            excitedClient.setCreditBalance(client.getCreditBalance());
-        }
-        return clientRepository.save(client);
+    @Transactional
+    public UpdateClientDTO update(UpdateClientDTO updateClientDTO, Long id) {
+        Client client = clientRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Client not found"));
+        client.setFirstName(updateClientDTO.getFirstName());
+        client.setLastName(updateClientDTO.getLastName());
+        client.setAge(updateClientDTO.getAge());
+        client.setLogin(updateClientDTO.getLogin());
+        client.setPassword(updateClientDTO.getPassword());
+        client.setCreditBalance(updateClientDTO.getCreditBalance());
+        Client savedClient = clientRepository.save(client);
+        UpdateClientDTO dto = new UpdateClientDTO();
+        dto.setId(savedClient.getId());
+        dto.setFirstName(savedClient.getFirstName());
+        dto.setLastName(savedClient.getLastName());
+        dto.setAge(savedClient.getAge());
+        dto.setLogin(savedClient.getLogin());
+        dto.setPassword(savedClient.getPassword());
+        dto.setCreditBalance(savedClient.getCreditBalance());
+        return dto;
     }
 
     @Override
     public void delete(Long id) {
         clientRepository.deleteById(id);
-
     }
 
     @Override
-    public List<Client> getAll() {
-        return clientRepository.findAll();
+    public List<GetAllClientsDTO> getAll() {
+        List<Client> clientList = clientRepository.findAll();
+        return clientList.stream().map(client -> {
+            GetAllClientsDTO getAllClientsDTO = new GetAllClientsDTO();
+            getAllClientsDTO.setId(client.getId());
+            getAllClientsDTO.setFirstName(client.getFirstName());
+            getAllClientsDTO.setLastName(client.getLastName());
+            getAllClientsDTO.setAge(client.getAge());
+            getAllClientsDTO.setLogin(client.getLogin());
+            getAllClientsDTO.setPassword(client.getPassword());
+            getAllClientsDTO.setCreditBalance(client.getCreditBalance());
+            return getAllClientsDTO;
+        }).collect(Collectors.toList());
     }
-
 }
```

Comment:
Do a research: Can we generify all of our mappings?

After all coments plz read about DataFaker library